api/interservice/authz/v2/project.proto

```diff
@@ -111,6 +111,8 @@ message ProjectRule {
     string name = 3;
     ProjectRuleTypes type = 4;
     repeated Condition conditions = 5;
+    bool deleted = 6;
+    string status = 7;
```

Comment:
_OMG why no enum?!!!_

I hear you -- but it's just so annoying to convert three different kinds of enums (storage layer, authz-service GRPC, automate-gateway GRPC), and I'm not convinced it's worth it. Happy to listen to arguments, but maybe we can just make our lives a little brighter and our PRs a little smaller. 😉 

Also, in the external API (automate-gateway), we can (and should) still use an enum.

Reply:
I really like the compromise of having an enum at the gateway and not in the domain layer. The enum unwrapping code is super super annoying.

Reply:
I really like the compromise of having an enum at the gateway and not in the domain layer. The enum unwrapping code is super super annoying.